docs/spec/v2alpha1/helmreleases.md

```diff
@@ -0,0 +1,276 @@
+# Helm Releases
+
+// TODO(hidde): short conclusive API description / outline
+
+## Specification
+
+A **helmrelease** object defines the source of a Helm chart by referencing an object
+managed by [source-controller](https://github.com/fluxcd/source-controller), the interval
+reconciliation should happen at, and a set of options to control the settings of the
+automated Helm actions that are being performed.
+
+```go
+type HelmReleaseSpec struct {
+	// ReleaseName used for the Helm release. Defaults to a composition of
+	// '[TargetNamespace-]Name'.
```

Comment:
Why add the `TargetNamespace`? If it's to ensure prevent HelmReleases accidentally using the same release name when setting `TargetNamespace`, then wouldn't it make more sense to just always default to `metadata.namespace-metadata.name` of the HelmRelease since that is always unique to a HelmRelease? Whereas two HelmReleases could choose the same `TargetNamespace`. I wonder if this is even needed though, especially if we're keeping something like the [antecedent annotation](https://docs.fluxcd.io/projects/helm-operator/en/stable/helmrelease-guide/reconciliation-and-upgrades/#the-antecedent-annotation) implemented.

Reply:
It is to prevent `HelmRelease` resources from colliding, see the explanation [here](https://docs.fluxcd.io/projects/helm-operator/en/stable/helmrelease-guide/release-configuration/#specifying-a-release-name).

> I wonder if this is even needed though, especially if we're keeping something like the antecedent annotation implemented.

We no longer annotate resources, nor do we try to be smart with Helm state (e.g. dry-runs). We simply record the revision of the latest Helm release as deployed by the controller in the status object of the resource and use this to determine if we need to perform an upgrade.